plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPageSink.java

```diff
@@ -111,10 +112,10 @@ public JdbcPageSink(ConnectorSession session, JdbcOutputTableHandle handle, Jdbc
                     .collect(toImmutableList());
         }
 
-        String sinkSql = getSinkSql(jdbcClient, handle, columnWriters);
         try {
-            sinkSql = remoteQueryModifier.apply(session, sinkSql);
-            statement = connection.prepareStatement(sinkSql);
+            String sql = sinkSql.orElseGet(() -> jdbcClient.buildInsertSql(handle, columnWriters));
```

Comment:
How about instead of injecting `Optional<String>` to inject `SinkSqlProvider`, where
```
interface SinkSqlProvider {
  String getSinkSql(JdbcClient jdbcClient, JdbcOutputTableHandle outputTableHandle, List<WriteFunction> columnWriters);
}
```

Reply:
Good idea! let me have a try~